tests/config/test_document_types_config.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+
+from briefcase.config import (
+    validate_document_description,
+    validate_document_ext,
+    validate_document_icon,
+    validate_document_url,
+)
+from briefcase.exceptions import BriefcaseConfigError
+
+
+@pytest.mark.parametrize(
+    "validation",
+    [
+        validate_document_icon,
+        validate_document_description,
+        validate_document_url,
+        validate_document_ext,
+    ],
+)
+def test_validations(validation):
+    valid_document = {
+        "icon": "icon",
+        "description": "description",
+        "url": "https://testurl.com",
+        "extension": "ext",
+    }
+    # Success does not raise an exception
+    validation("ext", valid_document)
+
+
+def test_validate_document_missing_field():
+    invalid_document = {
+        "description": "description",
+        "url": "https://testurl.com",
+        "extension": "ext",
+    }
+    # Failure raises an exception
+    with pytest.raises(BriefcaseConfigError):
```

Comment:
As before - adding a `match` clause is helpful here (and in other `raises` tests in this file)